consensus/istanbul/uptime/uptime.go

```diff
@@ -164,7 +156,8 @@ func (um *Monitor) ProcessBlock(block *types.Block) error {
 	// We only update the uptime for blocks which are greater than the last block we saw.
 	// This ensures that we do not count the same block twice for any reason.
 	if uptime == nil || uptime.LatestBlock < block.NumberU64() {
-		uptime = updateUptime(uptime, block.NumberU64(), signedValidatorsBitmap, um.lookbackWindow, epochNum, um.epochSize)
+		uptime = updateUptime(uptime, block.NumberU64()-1, signedValidatorsBitmap, um.lookbackWindow, um.MonitoringWindow(epochNum))
+		uptime.LatestBlock = block.NumberU64()
```

Comment:
Downside of doing this is here is the tests can't cover `LatestBlock`, but I guess that's not too bad.  The reason for moving it here was because of the `-1`?

Reply:
The `-1` and the fact that here is here it's actually used:

```
if latestBlock < currentBlock {
  do something
  latestBlock = currentblock
}  
```